### PR TITLE
Clean up see also for some reference pages

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

This attempts to follow the See also guideline to clean up some reference pages.

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/javascript/reference/classes/extends/index.md [hunk 1]
```diff
@@ -342,7 +342,8 @@ In this case, the `ReadOnlyMap` class is not a subclass of `Map`, but it still i
...
 ## See also
 
+- [Using classes](/en-US/docs/Web/JavaScript/Guide/Using_Classes)
 - [Classes](/en-US/docs/Web/JavaScript/Reference/Classes)
-- [`constructor`](/en-US/docs/Web/JavaScript/Reference/Classes/constructor)
-- [`super`](/en-US/docs/Web/JavaScript/Reference/Operators/super)
-- [Anurag Majumdar - Super & Extends in JavaScript](https://medium.com/beginners-guide-to-mobile-web-development/super-and-extends-in-javascript-es6-understanding-the-tough-parts-6120372d3420)
+- {{jsxref("Classes/constructor", "constructor")}}
+- {{jsxref("Statements/class", "class")}}
+- {{jsxref("Operators/super", "super")}}
```
Comment (dipikabh) [inline files/en-us/web/javascript/reference/classes/extends/index.md:349]: same comment here

File: files/en-us/web/javascript/reference/classes/private_class_fields/index.md [hunk 2]
```diff
@@ -370,7 +370,9 @@ PrivateConstructor.create(); // PrivateConstructor {}
 ## See also
 
 - [Using classes](/en-US/docs/Web/JavaScript/Guide/Using_Classes)
+- [Classes](/en-US/docs/Web/JavaScript/Reference/Classes)
 - [Public class fields](/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields)
+- {{jsxref("Statements/class", "class")}}
 - [Private Syntax FAQ](https://github.com/tc39/proposal-class-fields/blob/main/PRIVATE_SYNTAX_FAQ.md)
```
Comment (dipikabh) [inline files/en-us/web/javascript/reference/classes/private_class_fields/index.md:376]: same comment here - jsxref, then Reference links, then Guide link.

Also, could we say "Private syntax FAQ on tc39 on Github"?

**Review by dipikabh — COMMENTED** [review]
Looked through a few files and I've added suggestions to reorder the items in the list

**Comment by Josh-Cena** [comment]
@dipikabh I'm not sure about ordering links as "jsxref > Reference > Guide".

I have deliberately made "Guide" the first link: every reference should map to one unique guide page, and if someone wants to get an overview of a set of APIs, the guide is the best place to go.

Jsxref vs. Markdown link seems like an implementation detail. There's nothing preventing me from using `{{jsxref("Functions", "Functions", "", 1)}}` or ``[`class`](/en-US/docs/Web/JavaScript/Reference/Statements/class)``, so I don't think this should determine their ordering. That said, I don't have any clear guidelines for how to order the links either. I'm going with "Parent page > Sibling pages > Other pages" at the moment.

**Pull request merged**